text/0000-format.md

```diff
@@ -0,0 +1,64 @@
+- Feature Name: format
+- Start Date: 2016-07-27
+- RFC PR: (leave this empty)
+- Pony Issue: (leave this empty)
+
+# Summary
+
+This RFC proposes the seperation of the `string()` method and formatting. This wil remove formatting from the builtin package of the standard library and seperate the disperate concerns of producing a string representation of an object and formatting strings.
+
+# Motivation
+
+String formatting should not be involved in the string representation of an object. Pony's string formatting should also have improved documentation since it differs from that of most other languages.
+
+# Detailed design
+
+- Redefine the Stringable interface:
+```pony
+interface box Stringable
+  """
+  Things that can be turned into a String.
+  """
+  fun string(): String
```

Comment:
Thinking more about it, it seems like `val` is more specialized - more often, `iso` will be appropriate since the string will be built from scratch on every call.  In this paradigm, the best way to avoid allocating multiple strings is simply to have them be held onto by the caller.

If a class had a strong use case for returning a `String val` representation, it could provide another method, which would not provide the `Stringable` interface.

So, I think `String iso` is indeed more appropriate here.